File: kanito/Classi/CustomControls/MyTextField/MyTextField.m
//
//  MyTextField
//  NextGen
//
//  Created by Lc
//  Copyright (c) 2016 Lc. All rights reserved.
//

#import "MyTextField.h"

@interface MyTextField ()
@property (nonatomic, assign) CGFloat oldYposition;
@end

@implementation MyTextField

#pragma mark - Initialization / dealloc

- (instancetype)init {
    self = [super init];
    
    if(self) {
        [self initialization];
    }
    
    return self;
}

- (instancetype)initWithCoder:(NSCoder *)aDecoder {
    self = [super initWithCoder:aDecoder];
    
    if(self) {
        [self initialization];
    }
    
    return self;
}

- (instancetype)initWithFrame:(CGRect)frame {
    self = [super initWithFrame:frame];
    
    if(self) {
        [self initialization];
    }
    
    return self;
}

#pragma mark - Protected methods

- (void)initialization {
}

@end

File: Swift/Cls/CustomClass/MYTextField.swift
//
//  MYTextField
//  Enci
//
//  Created by Luciano Calderano on 03/11/16.
//  Copyright © 2016 Kanito. All rights reserved.
//

import UIKit

class MYTextField: UITextField {
    
    @IBOutlet var nextTextField: MYTextField?
    
    required internal init?(coder aDecoder: NSCoder) {
        super.init(coder: aDecoder)
        self.initialize()
    }
    
    override init(frame: CGRect) {
        super.init(frame: frame)
        self.initialize()
    }
    override internal func awakeFromNib() {
        super.awakeFromNib()
        self.initialize()
    }
    
    private func initialize () {
        self.font = UIFont.mySize(16)
//        self.layer.borderColor = UIColor.myBlueLight.cgColor
//        self.layer.borderWidth = 1
//        self.layer.cornerRadius = 5
    }

    func showError () {
        self.becomeFirstResponder()
//        self.layer.borderColor = UIColor.myRed().cgColor
//        delayWithSeconds(2) {
//            self.layer.borderColor = UIColor.myBlue().cgColor
//        }
    }
    
    private func delayWithSeconds(_ seconds: Double, completion: @escaping () -> ()) {
        DispatchQueue.main.asyncAfter(deadline: .now() + seconds) {
            completion()
        }
    }
}
